bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImplTest.java

```diff
@@ -127,6 +127,7 @@ public void setUp() throws Exception {
         Utils.setupHttpResponse("graphql/magento-graphql-attributes-result.json", httpClient, HttpStatus.SC_OK, "{customAttributeMetadata");
         Utils.setupHttpResponse("graphql/magento-graphql-category-result.json", httpClient, HttpStatus.SC_OK, "{category");
         Utils.setupHttpResponse("graphql/magento-graphql-search-result.json", httpClient, HttpStatus.SC_OK, "{products");
+        Utils.setupHttpResponse("graphql/magento-graphql-search-result-with-category.json", httpClient, HttpStatus.SC_OK, "{products");
```

Comment:
These two rules are conflicting, they share the same prefix `{products`. I'm not sure what happens then, I mean which response you get. Actually I think you probably no longer need the previous lines with `magento-graphql-search-result.json` and `magento-graphql-category-result.json` because this is no longer needed (the components now does a single query for both = the new file you added). If these two files are no longer needed, please also remove them.

I think you should also extend the test to verify that the GraphQL client does the number of queries you expect (that is, one).

Reply:
My bad, forgot to use `setupHttpResponse(String filename, HttpClient httpClient, int httpCode, String startsWith, String contains)` defined in `Utils.java`.
Will fix this

Reply:
Not sure you need this, because IIUC, you don't need the two previous response files any more.